File: inventory_stack/lib/core/models/item.dart
import 'package:inventory_stack/core/models/item_type.dart';
import 'package:inventory_stack/core/models/place.dart';
import 'package:json_annotation/json_annotation.dart';

part 'item.g.dart';

@JsonSerializable()
class ItemData {
  @JsonKey(name: "create_at")
  final String createAt;
  @JsonKey(name: "upgrade_at")
  final String upgradeAt;
  @JsonKey(name: "delete_at")
  final String deleteAt;
  final String uuid;
  @JsonKey(name: "serial_number")
  final String serialNumber;
  @JsonKey(name: "reg_number")
  final String regNumber;
  @JsonKey(name: "internal_number")
  final String internalNumber;
  final String name;
  final String? description;
  final ItemTypeData type;
  final String? date;
  @JsonKey(name: "root_place")
  final PlaceData rootPlace;
  @JsonKey(name: "current_place")
  final PlaceData currentPlace;

  ItemData(this.createAt, this.upgradeAt, this.deleteAt, this.uuid, this.serialNumber, this.regNumber, this.internalNumber, this.name, this.description, this.type, this.date, this.rootPlace, this.currentPlace);

  factory ItemData.fromJson(Map<String, dynamic> json) => _$ItemDataFromJson(json);
  Map<String, dynamic> toJson() => _$ItemDataToJson(this);
}

File: data_parser/models/item.go
package models

import (
	"encoding/json"
	"strconv"
)

type StringInt string

func (cf *StringInt) UnmarshalJSON(b []byte) error {
	var item interface{}
    if err := json.Unmarshal(b, &item); err != nil {
        return err
    }
    switch v := item.(type) {
    case int:
        *cf = StringInt(strconv.Itoa(v))
    case float64:
        *cf = StringInt(strconv.Itoa(int(v)))
    case string:
        *cf = StringInt(v)

    }
	return nil
}

type Item struct {
	Id             int `json:"id"`
	UUID           string `json:"uuid"`
	InternalNumber StringInt `json:"internal_number,int"`
	RegNumber      string `json:"reg_number"`
	TypeUUID	   string `json:"type_uuid"`
	Type		   ItemType `json:"type"`
	Name           string `json:"name"`
	Description    string `json:"description"`
	SerialNumber   StringInt `json:"serial_number"`
	Date           string `json:"date"`
	Count          int `json:"count"`
}